alchemiscale/tests/unit/test_keyedchain.py

```diff
@@ -0,0 +1,46 @@
+from alchemiscale.keyedchain import KeyedChain
+from alchemiscale.utils import RegistryBackup
+
+from gufe.tokenization import get_all_gufe_objs, GufeTokenizable, is_gufe_key_dict
+
+
+def test_keyedchain_full_network(network):
+    objects = get_all_gufe_objs(network)
+
+    for o in objects:
+        with RegistryBackup():
+            kc = KeyedChain.from_gufe(o)
+            _o = kc.to_gufe()
+            assert o == _o
+
+
+def test_keyedchain_len(network):
+    objects = get_all_gufe_objs(network)
+    expect_len = len(objects)
+
+    keyedchain = KeyedChain.from_gufe(network)
+
+    assert len(keyedchain) == expect_len
+
+
+def test_keyedchain_get_keys(network):
+    keyedchain = KeyedChain.from_gufe(network)
+    keys = list(map(lambda x: x.key, get_all_gufe_objs(network)))
+
+    for key in keyedchain.gufe_keys():
+        assert key in keys
```

Comment:
Consider instead doing a `set` comparison, since as written this only checks that all keys in the `KeyedChain` are in `keys`, but not the inverse.